src/evict/evict_lru.c

```diff
@@ -773,14 +773,14 @@ __evict_clear_walk(WT_SESSION_IMPL *session, bool count_stat)
 	if ((ref = btree->evict_ref) == NULL)
 		return (0);
 
-	if (count_stat)
-		WT_STAT_CONN_INCR(session, cache_eviction_walks_abandoned);
+	WT_STAT_CONN_INCR(session, cache_eviction_walks_abandoned);
 
 	/*
-	 * Clear evict_ref first, in case releasing it forces eviction (we
-	 * assert we never try to evict the current eviction walk point).
+	 * Clear evict_ref before releasing it in case that forces eviction (we
+	 * assert that we never try to evict the current eviction walk point).
 	 */
-	btree->evict_ref = NULL;
+	WT_PUBLISH(btree->evict_ref, NULL);
```

Comment:
I disagree -- first up, why do you think any barrier is required here?

Second, I want to avoid sprinkling explicit barriers around, so some more general mechanism would be better (and at least the pairing of `WT_PUBLISH` and `WT_ORDERED_READ` has reasonable mapping into the C11 memory model).

Thirdly, a read barrier here implies that you're worried that *this* thread is going to read something from before it set `evict_ref`.  What reads are you concerned about ordering in that way?